gremlin-go/driver/resultSet.go

```diff
@@ -120,6 +121,19 @@ func (channelResultSet *channelResultSet) Close() {
 	}
 }
 
+// Close and remove from the channelResultSet from the container without locking container. Meant for use when calling
+// function already locks the container.
+func (channelResultSet *channelResultSet) unlockedClose() {
```

Comment:
what happens if 2 go routines this method simultaneously?

Reply:
I think the idea is that this should never be called outside of closeAll and closeAll uses a mutex to ensure there is no concurrent usage.

Reply:
it makes sense